torchao/prototype/qat/nvfp4.py

```diff
@@ -12,6 +14,41 @@
 )
 
 
+class _NVFP4FakeQuantize(torch.autograd.Function):
+    """
+    Fake quantize a high precision tensor to nvfp4 and back with backward STE.
+    """
+
+    @staticmethod
+    def forward(
+        ctx, x: torch.Tensor, per_tensor_scale: Optional[torch.Tensor]
+    ) -> torch.Tensor:
+        block_size = 16
+
+        # quantize
+        scale, q = _nvfp4_quantize(
```

Comment:
umm, how about returning `NVFP4Tensor` here (wrapped with STE), and then just dequantizing it?  then you don't need `skip_packing` flag at all, you don't need to care about `f4_unpacked_to_f32`, etc?